test/broadcast.jl

```diff
@@ -346,7 +346,7 @@ end
 let f17314 = x -> x < 0 ? false : x
     @test eltype(broadcast(f17314, 1:3)) === Int
     @test eltype(broadcast(f17314, -1:1)) === Integer
-    @test eltype(broadcast(f17314, Int[])) === Union{Bool,Int}
+    @test eltype(broadcast(f17314, Int[])) == Union{Bool,Int}
```

Comment:
It's a bug in the test. The type might be `Union{Bool,Int}` or `Union{Int,Bool}`, which have different structures that can be distinguished by `===`. The `show` method for unions sorts the types though, obscuring this. #22664 is about normalizing union types so this doesn't come up.